kpm/lib/kpm/diagnostic_file.rb

```diff
@@ -149,8 +156,24 @@ def get_log_files(log_dir)
       # Helpers
 
       def get_system_catalina_base(export_data)
+        @catalina_base = nil
         system_json = JSON.parse(export_data)
-        @catalina_base = system_json['java_system_information']['catalina.base']['value']
+
+        if system_json['java_system_information']['catalina.base'].nil?
+
+          # find jcmd
+          jcmd = ( ENV['JAVA_HOME'] || '/**' ) + File::Separator + 'bin' + File::Separator + 'jcmd'
```

Comment:
Let's move this logic to `system.rb`, so the base script can better find `jcmd` as well.